system/lib/wasmfs/backends/node_backend.cpp

```diff
@@ -217,19 +217,16 @@ class NodeDirectory : public Directory {
     }
   }
 
-  bool removeChild(const std::string& name) override {
+  int removeChild(const std::string& name) override {
     auto childPath = getChildPath(name);
     // Try both `unlink` and `rmdir`.
     if (auto err = _wasmfs_node_unlink(childPath.c_str())) {
       if (err == EISDIR) {
         err = _wasmfs_node_rmdir(childPath.c_str());
       }
-      if (err) {
-        // TODO: Report specific errors.
-        return false;
-      }
+      return err;
```

Comment:
library_wasmfs_node.js seems return positive error codes, so this needs to be negated.
```suggestion
      return -err;
```